dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+	            context.put("HTMLPAGE_KEYWORDS", htmlPage.getSeoKeywords());
+	            context.put("HTMLPAGE_SECURE", String.valueOf(htmlPage.isHttpsRequired()));
+	            context.put("HTMLPAGE_REDIRECT", htmlPage.getRedirect());
+	            context.put("friendlyName", htmlPage.getFriendlyName());
+	            context.put("pageTitle", htmlPage.getTitle());
+                context.put("dotPageMode", mode);
+	            Date moddate = htmlPage.getModDate();
+
+	            moddate = new Date(moddate.getTime());
+
+	            context.put("HTML_PAGE_LAST_MOD_DATE", moddate);
+
+	            try {
+	                context.put("htmlPageInode", htmlPage.getInode());
+
+	                // for browsing the tree
+	                String view = java.net.URLEncoder.encode("(working=" + com.dotmarketing.db.DbConnectionFactory.getDBTrue()
+	                        + " and deleted=" + com.dotmarketing.db.DbConnectionFactory.getDBFalse() + "and language_id = "
+	                        + (String) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE) + ")", "UTF-8");
+	                context.put("view", view);
+	            } catch (Exception e) {
+	                Logger.warn(VelocityUtil.class, e.toString(), e);
+	            }
+	        }
+
+	        context.put("HTMLPAGE_SERVER_NAME", request.getServerName());
+	        context.put("VTLSERVLET_URI", UtilMethods.encodeURIComponent(idURI));
+	        if (request.getQueryString() != null && request.getQueryString().length() > 0) {
+	            context.put("queryString", request.getQueryString());
+	        } else {
+	            context.put("queryString", "");
+	        }
+	        context.put("TEMPLATE_INODE", String.valueOf(cmsTemplateInode));
+
+	        context.put("mainFrame", request.getParameter("mainFrame"));
+	        context.put("previewFrame", request.getParameter("previewFrame"));
+	        context.put(mode.name(), true);
+	        if (mode == PageMode.EDIT_MODE) {
+	            // gets user id from request for mod user
+	            com.liferay.portal.model.User backendUser = null;
+
+	            try {
+	                backendUser = com.liferay.portal.util.PortalUtil.getUser(request);
+	                // Skin skin = backendUser.getSkin();
+	                // context.put("USER_SKIN", skin.getSkinId());
+	            } catch (Exception nsue) {
+	                Logger.warn(VelocityUtil.class, "Exception trying yo getUser: " + nsue.getMessage(), nsue);
+	            }
+
+	            // to check user has permission to publish this page
+	            boolean permission = APILocator.getPermissionAPI().doesUserHavePermission(htmlPage, PERMISSION_PUBLISH, backendUser);
+	            context.put("permission", new Boolean(permission));
+
+	            // Check if the user is a CMS Administrator
+	            boolean adminUser = false;
+	            try {
+
+
+	                String adminRoleKey = "";
+	                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)